File: regular/C# - 1/04. Console Input_Output/homeWork/04. Console Input and Output/06. Quadratic Equation/06. Quadratic Equation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _06.Quadratic_Equation
{
    class Program
    {
        static void Main(string[] args)
        {
            double a = double.Parse(Console.ReadLine());
            double b = double.Parse(Console.ReadLine());
            double c = double.Parse(Console.ReadLine());
            double result;

            double D = Math.Pow(b, 2) - (4 * a * c);

            if (D < 0)
            {
                Console.WriteLine("no real roots");
            }
            else if (D == 0)
            {
                result = (-b) / (2 * a);
                Console.WriteLine("{0:F2}", result);
            }
            else {
                result = ((-b) - Math.Sqrt(D)) / (2 * a);
                Console.WriteLine("{0:F2}", result);
                result = ((-b) + Math.Sqrt(D)) / (2 * a);
                Console.WriteLine("{0:F2}", result);
            }
        }
    }
}

File: regular/JavaScript/01. Fundamentals/03. Conditional Statements/homeWork/06. Quadratic equation.js
function quadraticE(args){
    var a = +args[0],
        b = +args[1],
        c = +args[2],
        d = (b * b) - (4 * a * c),
        sqrtOFd = Math.sqrt(d),
        x1, x2;

    if (d > 0) {
        
        x1 = (-b - sqrtOFd) / (2 * a);
        x2 = (-b + sqrtOFd) / (2 * a); 
        if (x1 > x2) {
            var temp = x1;
            x1 = x2;
            x2 = temp;
        }
        console.log('x1=' + x1.toFixed(2) + "; x2=" + x2.toFixed(2));
    }
    else if (d === 0) {
        x1 = -b / (2 * a);
        console.log('x1=x2='+ x1.toFixed(2));
    }
    else if(d < 0){
        console.log('no real roots');
    }
}